### PR TITLE
Crypto store and olm machine tests

### PR DESCRIPTION
**Pull request by nikofil**
* Add tests for the functionality of the Gob and SQL crypto stores
* Change `RemoveOutboundGroupSession` in the Gob store to not check if a session is shared, to be consistent with the SQL store
* Add a test for the functionality of `OlmMachine`, creating an olm and megolm session, sharing the megolm session and encrypting a message

### CODE REVIEW

**Comment by nikofil** [comment]
Hopefully these are useful, if you'd like tests for something else let me know.
I plan to look at some of the TODOs in the code next, particularly in the crypto part, if that doesn't conflict with what you're doing.

File: crypto/machine_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2020 Tulir Asokan
```
Comment (tulir) [inline crypto/machine_test.go:1]: You should put your name in things you wrote

Comment (nikofil) [reply crypto/machine_test.go]: Sure, changed

File: crypto/store_test.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	if err != nil {
+		t.Fatalf("Error opening db: %v", err)
+	}
+	sqlStore := NewSQLCryptoStore(db, "sqlite3", id.DeviceID("dev"), []byte("test"), emptyLogger{})
+	if err = sqlStore.CreateTables(); err != nil {
+		t.Fatalf("Error creating tables: %v", err)
+	}
+
+	os.Remove("gob_store_test.gob")
+	gobStore, err := NewGobStore("gob_store_test.gob")
```
Comment (tulir) [inline crypto/store_test.go:37]: This doesn't seem to get removed after the tests

Comment (nikofil) [reply crypto/store_test.go:37]: Right, the last test was leaving its gob. Fixed that by also returning a cleanup func from `getCryptoStores`.

**Pull request merged**